rita_common/src/tunnel_manager/neighbor_status.rs

```diff
@@ -18,46 +17,38 @@ pub fn get_neighbor_status() -> HashMap<Identity, NeighborStatus> {
     NEIGHBOR_STATUS.read().unwrap().clone()
 }
 
-pub struct UpdateNeighborStatus;
-
-impl Message for UpdateNeighborStatus {
-    type Result = ();
-}
-
-impl Handler<UpdateNeighborStatus> for TunnelManager {
-    type Result = ();
-
-    fn handle(&mut self, _: UpdateNeighborStatus, _: &mut Context<Self>) -> Self::Result {
-        let mut external_list = NEIGHBOR_STATUS.write().unwrap();
-        for (id, tunnel_list) in self.tunnels.iter() {
-            // we may have many tunnels with this same peer, we want to get
-            // the lowest shaper value of any of the recently active tunnels
-            let mut lowest_shaper_speed = None;
-            let mut enforced = false;
-            for tunnel in tunnel_list.iter() {
-                match (tunnel.speed_limit, lowest_shaper_speed) {
-                    (Some(new), Some(current)) => {
-                        if new < current {
-                            lowest_shaper_speed = Some(new);
-                        }
+/// Handles updates to neighbor status with lazy static lock
+pub fn update_neighbor_status(tunnel_manager: &mut TunnelManager) {
+    let mut external_list = NEIGHBOR_STATUS.write().unwrap();
```

Comment:
see this as an example for the scheme to avoid multiple .write() calls in the same function.